File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, a in enumerate(nums):
            b = target - a
            j = seen.get(b)
            if j is not None:
                return [i, j]
            seen[a] = i
# @lc code=end

print(Solution().twoSum([2, 9, 7, 11], 9))

File: 1.two-sum.cpp
#include <vector>
#include <unordered_map>

using namespace std;

/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(const vector<int>& nums, int target)
    {
        std::unordered_map<int, int> available;
        for (int i = 0; i < nums.size(); ++i) {
            int a = target - nums[i];
            auto it = available.find(a);
            if (it != available.end() && it->second != i) {
                return {i, it->second};
            }
            available[nums[i]] = i;
        }
        return {};
        
    }
};
// @lc code=end

